### ISSUE TITLE
Split tag initializers out of InitializeElement

### ISSUE THREAD

**Issue by nilsvu** [sxs-collaboration/spectre#996]
# Feature request:

`InitializeElement` is already nicely separated into initializers that divide up the responsibility for initializing different parts of the DataBox tags (thanks @wthrowe). They chain together, passing partially initialized DataBoxes down the line. Each initializer exposes `simple_tags`, `compute_tags` and a static `initialize` function.

We should take that architecture further and move some (or all) links of this chain out of the file. #792 does so already for the linear solver initialization for elliptic systems.

Examples:

- The `SystemTags` struct could be moved to `Evolution/{Non-}Conservative/InitializeTags.hpp`. Then each system exposes a type `system::tags` that would be set to, for instance, `Evolution::Conservative::InitializeTags` or `Evolution::NonConservative::InitializeTags`. In `InitializeElement` we need only add the `system::tags::simple_tags` and `system::tags::compute_tags` to the `return_tag_list` and call the `system::tags::initialize` function. This could replace the `is_conservative`, since it's currently only used for initialization.
- `EvolutionTags` -> `Evolution/InitializeTags.hpp`
- ...

### Component:

- [x] Code

**Comment by nilsdeppe** [comment]
@kidder and I were talking about this in the context of just the evolution and it'll be very useful, so this is planned but not something any of us can work on right now.

**Comment by kidder** [comment]
yes, I am writing some initializers now for GrMhd, so it will be better to wait until those are finished before coming up with a more generic structure.

**Comment by nilsvu** [comment]
Yep it's not urgent. I'll try to set up a structure like this also for elliptic systems and perhaps it'll have spin-offs for evolution code :)

**Comment by nilsvu** [comment]
#1207 follows the suggested structure

**Comment by kidder** [comment]
This is addressed by #1535 #1579 and #1575
